### PR TITLE
compute,storage: move variable_length_row_encoding flag into CreateTimely

### PR DESCRIPTION
**Pull request by teskje**
This PR moves the `variable_length_row_encoding` feature flag, which controls whether or not to enable variable-length row encoding, from `CreateInstance` into the `CreateTimely` command. In contrast to `CreateInstance`, `CreateTimely` is also part of the storage protocol. So the move ensures that the `variable_length_row_encoding` flag is     applied by both the compute and the storage parts of a replica.

Prior to this change, storage would be oblivious to the `variable_length_row_encoding` flag and would never enable row encoding. This could lead to a problematic sequence where storage would start up and begin hydrating a source, then later compute would process its `CreateInstance` command and change the global variable-length row encoding configuration. The storage source would start performing variable-length encoding mid-hydration, which would mess up its invariants and lead to panics.

The first commit of this PR contains some minor cleanup around compute's `InstanceConfig` type.

h/t to @guswynn [for coming up with the diagnosis](https://materializeinc.slack.com/archives/C01CFKM1QRF/p1693955892057169)!

### Motivation

  * This PR fixes a recognized bug.

Fixes https://github.com/MaterializeInc/materialize/issues/21605.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [x] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [x] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - N/A

### CODE REVIEW

**Review by guswynn — APPROVED** [review]
Thanks so much for getting this up @teskje!

File: src/cluster/src/server.rs [hunk 1]
```diff
@@ -237,22 +237,25 @@ where
                 existing
             }
             None => {
-                let build_timely_result = Self::build_timely(
+                // Configure variable-length row encoding.
```
[the remark this answers is not in the document]
Comment (guswynn) [reply src/cluster/src/server.rs:240]: (this is just a nit, not required)

**Comment by teskje** [comment]
TFTRs!

File: src/cluster/src/server.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (teskje) [reply src/cluster/src/server.rs:240]: Ah, I somehow didn't see that before merging, sorry! 🙈

**Pull request merged**